test/util/testnode/read.go

```diff
@@ -96,6 +99,67 @@ func reverseSlice[T any](s []T) {
 	}
 }
 
+type HeightTime struct {
+	Height int64
+	Time   time.Duration
+}
+
+func CalculateBlockTime(blockMetas []*types.BlockMeta) []HeightTime {
+	// sort by Height in ascending order
+	sort.Slice(blockMetas, func(i, j int) bool {
+		return blockMetas[i].Header.Height < blockMetas[j].Header.Height
+	})
+
+	var result []HeightTime
+	for i := 0; i < len(blockMetas)-1; i++ {
+		// the time difference between the current header and the next one
+		timeDiff := blockMetas[i+1].Header.Time.Sub(blockMetas[i].Header.Time)
+		// shift the time difference to the next height
+		result = append(result, HeightTime{Height: blockMetas[i+1].Header.Height, Time: timeDiff})
+	}
+	return result
+}
```

Comment:
This stuff is probably duplicated with the blocktime tool that I wrote but we can come back and clean that up later

Reply:
moved to a gist on this comment for now https://github.com/celestiaorg/celestia-app/issues/3965#issuecomment-2413746001